### PR TITLE
Assessments: Set up basic controller tests

### PR DESCRIPTION
**Pull request by caleybrock**
Early version of controller tests for assessments. This only includes authentication tests. I wanted to set these up before I start moving some logic out of the controllers and into the models.

### CODE REVIEW

File: dashboard/test/controllers/api/v1/assessments_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    sign_in non_verified_teacher
+    get :index
+    assert_response :not_found
+  end
+
+  test 'verified teacher can get assessment questions and answers' do
+    sign_in @teacher
+    get :index, params: {section_id: @section.id, script_id: 2}
+    assert_response :success
```
Comment (maddiedierker) [inline dashboard/test/controllers/api/v1/assessments_controller_test.rb:38]: i would add an assertion here to make sure you get the assessment(s) you expect from the index action

File: dashboard/test/controllers/api/v1/assessments_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  test 'students cannot get assessment responses from students' do
+    sign_in @student
+    get :section_responses
+    assert_response :forbidden
+  end
+
+  test 'verified teacher can get assessment responses from students' do
+    sign_in @teacher
+    get :section_responses, params: {section_id: @section.id, script_id: 2}
+    assert_response :success
```
Comment (maddiedierker) [inline dashboard/test/controllers/api/v1/assessments_controller_test.rb:57]: same as comment above - just make sure the correct section responses come back as well

**Comment by caleybrock** [comment]
@madelynkasula updated!

**Comment by maddiedierker** [comment]
@caleybrock i think having both endpoints return some data would be really useful for these tests. something like this:

https://github.com/code-dot-org/code-dot-org/blob/1c67af3e880d9568c7fb52a68714012091bfe8e1/dashboard/test/controllers/api/v1/sections_controller_test.rb#L46-L54

that file and the [courses_controller_test.rb](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/courses_controller_test.rb) have a lot of good examples of this

**Comment by caleybrock** [comment]
I totally agree. The goal is for these to come later, I just wanted to first ensure the authentication is correct first.

**Pull request merged**